### DISCUSSION TITLE
Pimcore Academy Pimcore Upgrade 6.9

### DISCUSSION THREAD

**Discussion by andi2802** [pimcore/pimcore#9279] [Q&A / Support] [answered]
Hello,
since 5 weeks I try to solve the question about Pimcore upgrade 6.9 in the Senior Developer Course. I went through everyting available in the course at least three times, I read anything I could find about changelogs or Upgrade Notes, but I can't pass this exam.

Is there maybe an error in the questions? Or what do I do wrong?
Thanks for any advice or hint. 

It is the only one I am still missing from Junior,Senior,Enterprise Developer and Consultant...

**Comment by dvesh3** [comment, marked as answer]
@andi2802 just go through the complete course '**What's new in Pimcore 6.9**', you'll find the correct answers.

**Reply by andi2802** [reply]
so a fourth time should finally make it clear? Ok I try it again next week...

**Reply by dvesh3** [reply]
Is there a specific answer that you are not sure about? I have confirmed that every information is provided in the course.

**Reply by andi2802** [reply]
There are several answers, where it says 'was removed', but in the course there stands will be removed. Is this the same? Or are these answers definitely wrong, because it is marked 'removed' and will be removed in Pimcore X?

**Reply by dvesh3** [reply]
Could you please screenshot of your answers? thanks!

**Reply by andi2802** [reply]
Here is a screenshot of one of my many tries.
<img width="1094" alt="Bildschirmfoto_6 9 2021-05-25 um 11 37 51" src="https://user-images.githubusercontent.com/60915613/120269369-f9ba5f80-c2a7-11eb-8d16-0d9ee2bd413e.png">

**Reply by dvesh3** [reply]
so let's review these answers:
 - Additional templating helper were introduced. **true**: https://github.com/pimcore/pimcore/pull/7463
 - Normalize/Denormalize will replace marshal/unmarshal of data object field definitions. **true**: https://github.com/pimcore/pimcore/issues/8487
 - Elasticsearch 7 support was introduced.. **true**: https://github.com/pimcore/pimcore/issues/7678
 - New settings store for simple key value data was introduced. **true**: https://github.com/pimcore/pimcore/pull/8332
 - Many to one relations are now allowed in reverse relations.. **true**: https://github.com/pimcore/pimcore/pull/8037
 - Geopoint data types were removed from data objects. **wrong** As Geopoint class was only deprecated in favor of GeoCoordinates class https://github.com/pimcore/pimcore/pull/8306
 
 I hope you can pass the exam now 😉

**Reply by andi2802** [reply]
Thank you!!

**Reply by betterapp** [reply]
@dvesh3 I have the same problem. You show the answer from git changes but the course does not have all answer. If yes - show us where is it :P

For example. Where in course I can find information that:
Normalize/Denormalize will replace marshal/unmarshal of data object field definitions - is true ??

Show us :P

**Reply by fashxp** [reply]
Hi, thx for the feedback. We added an additional note to the course: 

![image](https://user-images.githubusercontent.com/8792145/148203879-83c16bea-ca23-4449-81e5-7d6b602845b4.png)

**Reply by betterapp** [reply]
The whole answers can be found in Milestones for PimCore Versions.
Other course (for every version) does not have all information.

I have a proposition to add github link for milestones to the course for every chapter.
